### PR TITLE
Split creation of the Set to match unknown properties to avoid MethodTooLargeException

### LINKED ISSUE #44599: Build fails with "Method too large" exception
### Describe the bug

I have an application, which uses a lot (94) of different extensions. When I am trying to build it using the current SNAPSHOT, it fails with an exception during the build phase. If I use Quarkus 3.16.3, the apps works without any problems, same happens if I remove `quarkus-redis-client` extension

### Expected behavior

Application either compiles or fails with a more telling error

### Actual behavior

```
[error]: Build step io.quarkus.deployment.steps.ConfigGenerationBuildStep#generateConfigClass threw an exception: org.objectweb.asm.MethodTooLargeException: Method too large: io/quarkus/runtime/generated/Config.generateMappedProperties ()Ljava/util/Set;
[ERROR] 	at org.objectweb.asm.MethodWriter.computeMethodInfoSize(MethodWriter.java:2088)
[ERROR] 	at org.objectweb.asm.ClassWriter.toByteArray(ClassWriter.java:506)
[ERROR] 	at io.quarkus.gizmo.ClassCreator.writeTo(ClassCreator.java:246)
[ERROR] 	at io.quarkus.gizmo.ClassCreator.close(ClassCreator.java:257)
[ERROR] 	at io.quarkus.deployment.configuration.RunTimeConfigurationGenerator$GenerateOperation.run(RunTimeConfigurationGenerator.java:519)
[ERROR] 	at io.quarkus.deployment.steps.ConfigGenerationBuildStep.generateConfigClass(ConfigGenerationBuildStep.java:318)
[ERROR] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:499)
```

### How to Reproduce?

1. `git clone git@github.com:quarkus-qe/beefy-scenarios.git && cd beefy-scenarios/002-quarkus-all-extensions`
2. `mvn clean verify # or mvn clean install -DskipTests`

### Output of `uname -a` or `ver`

6.11.7-300.fc41.x86_64

### Output of `java -version`

21.0.1, vendor: Eclipse Adoptium

### Quarkus version or git rev

26db59d0b672053df63fd5c058b0778ffaac4739

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #44599

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java [hunk 1]
```diff
@@ -628,27 +628,27 @@ ReadResult run() {
                 objectsByClass.put(mapping.getKlass(), config.getConfigMapping(mapping.getKlass(), mapping.getPrefix()));
             }
 
-            Set<PropertyName> buildTimeNames = getMappingsNames(buildTimeMappings);
-            Set<PropertyName> buildTimeRunTimeNames = getMappingsNames(buildTimeRunTimeMappings);
-            Set<PropertyName> runTimeNames = getMappingsNames(runTimeMappings);
+            Map<PropertyName, String> buildTimeNames = mappingsToNames(buildTimeMappings);
+            Map<PropertyName, String> buildTimeRunTimeNames = mappingsToNames(buildTimeRunTimeMappings);
+            Map<PropertyName, String> runTimeNames = mappingsToNames(runTimeMappings);
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java:633]: Could you clarify why this change? It doesn't look related to splitting the method?

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java:633]: I dunno. It makes the intent less clear and I don't think you win anything by doing that, at least not a lot. I think what @franz1981 wanted to underline is that the keySet keeps the map around but given you switched to referencing the map directly, the only win is the very small keyset allocation. Given this is build time, I don't see the value of making the code more convoluted.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@radcortez all the reports were with `main`, right? Or do we try to backport it to 3.17 too?

**Comment by radcortez** [comment]
All `main` yes, but this was caused primarily by https://github.com/quarkusio/quarkus/pull/44079 which is set to 3.18, but also https://github.com/quarkusio/quarkus/pull/44448, which is set to 3.17.

It probably wouldn't hurt to backport, just in case.

**Pull request merged**